### DISCUSSION TITLE
Multiple network connections simultaneously

### DISCUSSION THREAD

**Discussion by I0San** [wevm/wagmi#1912] [Q&A] [answered]
Hi,

Is it possible to have a connection to different networks simultaneously? No wallet connected.

For example, if a DApp wants to read public properties or listen to events from multiple smart contracts on different networks.

Thanks.

**Comment by tmm** [comment, marked as answer]
Yes, check out the `chainId` hook property. It allows you to target configured chains.

```ts
import { useContractRead } from 'wagmi'

const { data } = useContractRead({ chainId: 10 })
```

**Reply by I0San** [reply]
That makes sense! Thanks!

One more thing, about alchemy and infura providers. You can add one API key per provider, but that API key is for one network usually. Can I add multiple alchemy providers with different keys for each network or will the "chainId" prop allow me to read from different networks through single alchemy or infura provider?

Thanks in advance.

**Comment by kombos** [comment]
@tmm  @I0San 
is there simultaneous multiple network support during writing of transactions? Or do we need to first perform switchNetwork() to the desired chain and then create and send transaction? the problem with switchNetwork is that several prepareContract functions that were configured to previous chain, starts throwing error the minute the network switches. also is there a way to get the provider (using connector.getProvider({chainId}) for a specific network/chain? giving chainId value is not working and its still sourcing from the connected network.

**Reply by I0San** [reply]
I did solve it by including chain id to read from different chains. To write, of course, you need to switch before and I did just that. Connect if not connected and then switch if on the wrong network, then write.
